### ISSUE TITLE
on cluster edit, machineConfig is not `PUT`, therefore updates are not applying properly

### ISSUE THREAD

**Issue by slickwarren** [rancher/dashboard#3553]
on master-head (1adc8ef)

Issue:
If a user edits a cluster (i.e. cluster region or subnet), the change is not being applied -> no `machineConfig` request is made

Repro Steps:
* create an ec2 single node cluster in us-west-2 region
* allow cluster to go to active, then edit the region to us-east-2
* cluster will remain active, and a request is sent for 1 object, but not for `machineConfig` therefore no change is actually made.

**Comment by gaktive** [comment]
Curious to see if this is also happening within the same region.

**Comment by vincent99** [comment]
Nothing to do with regions.. [saveMachinePools()](https://github.com/rancher/dashboard/blob/master/edit/provisioning.cattle.io.cluster/rke2.vue#L730-L735) only calls save for ones that are being created, it's missing handling updates.

**Comment by aaronRancher** [comment]
Tested on Rancher `v2.6-head` `version 222dced` with the following steps:
create an ec2 single node cluster in us-west-2 region
allow cluster to go to active, then edit the region to us-east-2 and choose a subnet in the same region.

Result:
The cluster remains active and does not go into updating state.
A machine config request is made with the updated region and subnet.
`https://.../v1/rke-machine-config.cattle.io.amazonec2configs/fleet-default/nc-ec2-cluster-<node pool name>-924f6`
```
region: "us-east-2"
requestSpotInstance: false
retries: "5"
rootSize: "16"
secretKey: null
securityGroup: ["rancher-nodes"]
0: "rancher-nodes"
securityGroupReadonly: false
sessionToken: null
spotPrice: "0.50"
sshKeyContents: null
sshUser: "ubuntu"
subnetId: "subnet-...110c"
```
The node pool is not recreated in the us-east-2 region in EC2 backend. Editing Rancher will show the node pool is in us-east-2 region.
![Screen Shot 2021-08-04 at 5 28 52 PM](https://user-images.githubusercontent.com/57409052/128272626-8cd430be-14b6-47c5-82d5-d278fcbe37e8.png)

**Comment by vincent99** [comment]
If the machine config is getting updated then that's backend or we need to start disabling things you can't really edit.  Editing region does sound like something that's not going to work very well, e.g. the networks are per-region.

**Comment by codyrancher** [comment]
@vincent99 your comments make me think that you expect to be ale to edit the properties of the machine pools and have those edits be reflected in the existing cluster. Is this new funactionality? 

In ember it's possible to switch the node template (effectively a maching pool) but when you save, nothing actually changes about the cluster. You can also edit a node template separately but none of the clusters that use the template actually change.

I personally think we should go back to be able to create/delete machine pools and disable editing. It looks to me that the backend doesn't support editing.

**Comment by vincent99** [comment]
That is the API I'd like, but not what we have discovered to be the API we have.

So I agree the machineconfig section of the pool should be disabled once the configure is saved* and should say to make a new pool if you want to change it for now.  Things like the quantity that go into the cluster spec should still be ok to edit.

*: this can happen even in create cluster, if you click edit as yaml we fire hooks and save the configs so the cluster will be filled or properly.  (And that now works as of last night, save wasn't actually waiting for hooks to finish before).

**Comment by rak-phillip** [comment]
Just to make sure I understand everything we need to change in order to move this one forward:

1. Revert change that attempts to the update machine config
2. Disable all fields under the Machine Pools section
3. Add a message to Machine Pools instructing the user to create a new cluster when attempting to edit
4. Potentially disable editing of machine config in yaml after saving configs when edit as yaml is clicked

**Comment by vincent99** [comment]
For 2, just the part that writes to the `entry.config`.  Things like the replica count are still ok because they are not part of the config.

Yes for 4; after you click create or edit yaml they are saved, get a name/id and the create bit is set to false so the changes for 2 should probably do this for "free".  (For the create button, there could be a validation error that leaves you on this screen instead of going back to the list).

**Comment by aaronRancher** [comment]
Tested on Rancher `v2.6.0-rc2`.
Machine Pool fields are disabled. There is a message instructing users to create a new machine pool on edit: `Create a new pool to update machine configs`. The user also cannot edit machine pool config from YAML.
![Screen Shot 2021-08-10 at 4 25 11 PM](https://user-images.githubusercontent.com/57409052/128947452-96d78170-4cf1-4bb6-af06-80c6e0ffbb78.png)
The user is unable to edit fields such as region or subnet.
But it is not clear at a glance these fields are disabled.

Reopening to make it clearer these fields are disabled.

**Comment by aaronRancher** [comment]
Will track disabled styling issue here: https://github.com/rancher/dashboard/issues/3797
Closing this, as the original issue has been fixed.

**Comment by johnwc** [comment]
Is this something new that has not been released? We're using v2.6.4 and I'm not see that behavior in the cluster edit UI. 

We updated our VM template in vSphere, which caused the (3) that was suffixed at the end of the template name to be incremented to (4). Now we can't increase our node count because it's pointing to a non-existing template name. I came to the cluster config to edit it to the new template name, but it will not update at all. How do I update the template in the cluster config? We're not expecting it to redeploy nodes with the new template, just use the new template on new nodes.

<img width="1213" alt="image" src="https://user-images.githubusercontent.com/2798441/169681721-195e3712-7a24-4584-882a-16914afe2e30.png">
